### PR TITLE
Add/Improve docs for config settings: "ignore", "ignore_templated_areas"

### LINKED ISSUE #831: Templated jinja is not ignored
When trying to use macros from a package, sqlfluff returns an error when these should be skipped because of the `ignore_templated_areas` parameter.

## Expected Behaviour

`dbt_utils` macros should be ignored in the lint command.

## Observed Behaviour

`sqlfluff lint` output

```
L:   - | P:   - |  TMP | Unrecoverable failure in Jinja templating: 'dbt_utils'
                       | is undefined. Have you configured your variables?
                       | https://docs.sqlfluff.com/en/latest/configuration.html
L:   4 | P:   4 |  TMP | Undefined jinja template variable: 'dbt_utils'
```
-- If this is a parsing or linting issue, please include a minimal SQL example which reproduces the issue, along with the `sqlfluff parse` output, `sqlfluff lint` output and `sqlfluff fix` output when relevant. Finally, please add any labels that are related to the issue before submitting so that it can be addressed easily! --

## Steps to Reproduce

Query:
```sql
select id
    , count(*)
from {{ source("snowplow", "events") }}
{{ dbt_utils.group_by(1) }}
```

Try to lint using:
> sqlfluff lint models/snowplow_events.sql
output:
```
== [models/snowplow_events.sql] FAIL
L:   - | P:   - |  TMP | Unrecoverable failure in Jinja templating: 'dbt_utils'
                       | is undefined. Have you configured your variables?
                       | https://docs.sqlfluff.com/en/latest/configuration.html
L:   4 | P:   4 |  TMP | Undefined jinja template variable: 'dbt_utils'
```

## Version
Include the output of `sqlfluff --version` along with your Python version

python 3.8.3

```
> sqlfluff --version
sqlfluff, version 0.4.1
```

## Configuration
```
[sqlfluff]
templater = jinja
ignore_templated_areas = True
```

### PR DESCRIPTION
**Pull request by barrywhart**
### Brief summary of the change made
Fixes #831

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -699,6 +699,16 @@ ignored until a corresponding `-- noqa:enable=<rule>[,...] | all` directive.
...
 
+Ignoring types of errors
+^^^^^^^^^^^^^^^^^^^^^^^^
+General *categories* of errors can be ignored using the ``--ignore`` command
+line option or the ``ignore`` setting in ``.sqlfluffignore``. Types of errors
+that can be ignored include:
+
+* ``lexing``
+* ``parsing``
+* ``templating``
```
Comment (tunetheweb) [inline docs/source/configuration.rst:710]: You can actually also ignore all `linting` errors too!

```suggestion
* ``lexing``
* ``linting``
* ``parsing``
* ``templating``
```

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/sqlfluff/core/default_config.cfg [hunk 2]
```diff
@@ -18,7 +18,11 @@ recurse = 0
 output_line_length = 80
 # Number of passes to run before admitting defeat
 runaway_limit = 10
-# Ignore linting errors in templated sections
+# Ignore errors by category.
+#ignore = lexing,parsing,templating
```
Comment (tunetheweb) [inline src/sqlfluff/core/default_config.cfg:22]:
```suggestion
#ignore = lexing,linting,parsing,templating
```

File: docs/source/configuration.rst (hunk 1 above)
Comment (barrywhart) [reply docs/source/configuration.rst:710]: Yes. I was wondering if we wanted to advertise that. I'll make the change. Are you aware of users wanting to do this?

File: src/sqlfluff/core/default_config.cfg (hunk 2 above)
Comment (barrywhart) [reply src/sqlfluff/core/default_config.cfg:22]: Sure.

File: docs/source/configuration.rst (hunk 1 above)
Comment (tunetheweb) [reply docs/source/configuration.rst]: Nope! But not aware of anyone regularly wanting to do the others either (and wouldn't advise it! It suggests something more fundamental is wrong).

Of all of those, ignoring linting errors is the LEAST likely to cause real problems IMHO 😁 Though, at the same time, it also negates much of the benefit of using SQLFluff!

File: src/sqlfluff/core/default_config.cfg [hunk 3]
```diff
@@ -18,7 +18,11 @@ recurse = 0
 output_line_length = 80
 # Number of passes to run before admitting defeat
 runaway_limit = 10
-# Ignore linting errors in templated sections
+# Ignore errors by category.
```
Comment (tunetheweb) [inline src/sqlfluff/core/default_config.cfg:21]:
```suggestion
# Ignore errors by category (one or more of the following, separated by commas: lexing,linting,parsing,templating)
```

**Review by tunetheweb — APPROVED** [review]

**Pull request closed without merging**